### PR TITLE
UI-491 AutoAlign

### PR DESCRIPTION
**Pull request by hanzge**
- Auto & manually fix eslint errors
- Add `defaultProps`
- Remove inline styles ~~and create a separate `scss` file~~
- Add BEM style `className`

### CODE REVIEW

File: lib/components/utils/AutoAlign.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.AutoAlign {
```
Comment (sheela-svmx) [inline lib/components/utils/AutoAlign.scss]: I am a bit confused. so if the whole autoALign is positioned absoluteley here? In the old code,there is some portaly class and only that had this absolute position. I am trying to understand and make sure this does not break when this autoalign is used elsewhere

File: lib/components/utils/AutoAlign.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  position: 'absolute';
+  left: 0;
```
Comment (sheela-svmx) [inline lib/components/utils/AutoAlign.scss]: I am a bit confused. so if the whole autoALign is positioned absoluteley here? In the old code,there is some portaly class and only that had this absolute position. I am trying to understand and make sure this does not break when this autoalign is used elsewhere

Comment (hanzge) [reply lib/components/utils/AutoAlign.scss]: I moved the inline style from the original code to this `scss` file. After I saw your comment, I tried to remove the `.AutoAlign` style and it still worked. So this file can be actually removed.

I did some further research. This HOC is using `react-relative-portal`. I looked up the [source code](https://github.com/smartprogress/react-relative-portal/blob/master/src/Portal.js#L59) and find that it's using 
`ReactDOM.unstable_renderSubtreeIntoContainer`. So `AutoAlign` is actually rendering elements into another React root node which is out side the main root node. The `.AutoAlign` class is on that new root node.

**Comment by joshsylvester** [comment]
two approvals, merged. 

published. {{@svmx/ui-components-lightning@0.1.31}}

**Pull request merged**